docs/bundles/SyliusGridBundle/custom_action.rst

```diff
@@ -0,0 +1,50 @@
+Custom Action
+=============
+
+There are certain cases when built-in action types are not enough.
+
+All you need to do is create your own action template and register it for the ``sylius_grid``.
+
+In the template we will specify the button;s icon to be ``mail`` and its colour to be ``purple``.
+
+.. code-block:: twig
+
+    {% import '@SyliusUi/Macro/buttons.html.twig' as buttons %}
+
+    {% set path = options.link.url|default(path(options.link.route)) %}
+
+    {{ buttons.default(path, action.label, null, 'mail', 'purple') }}
+
+Now configure the new action's template like below in the ``app/config/config.yml``:
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    sylius_grid:
+        templates:
+            action:
+                contactSupplier: "AppBundle:Grid/Action:contactSupplier.html.twig"
+
+From now on you can use your new action type in the grid configuration!
+
+Let's assume that you already have a route for contacting your suppliers, let's configure the grid action:
```

Comment:
`Let's` is twice in one sentence which sounds weird, at least to me. Maybe `, then you can configure...`?